src/components/Footer.tsx

```diff
@@ -0,0 +1,85 @@
+import { deleteTodo } from '../api/todos';
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+
+interface FooterProps {
+  todos: Todo[],
+  filter: Filter,
+  handleFilter: (filter: Filter) => void,
+  handleError: (error: string) => void,
+  handleIsUpdating: (status: boolean) => void,
+  handleUpdatingIds: (ids: number[]) => void,
+}
+
+export const Footer = ({
+  todos, filter, handleFilter, handleError, handleIsUpdating, handleUpdatingIds,
+}: FooterProps) => {
+  const hasCompleted = todos.some(todo => todo.completed);
```

Comment:
You can compare todosLeft and todos length, if it's different - we have completed todos. So you can get this value without iteration